File: ru.kmorozov.library.data.server/src/main/java/ru/kmorozov/library/data/server/options/ServerGBDOptions.java
package ru.kmorozov.library.data.server.options;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.context.annotation.Lazy;
import org.springframework.stereotype.Component;
import ru.kmorozov.gbd.core.config.IGBDOptions;
import ru.kmorozov.gbd.core.config.IStorage;
import ru.kmorozov.gbd.core.config.options.CtxOptions;
import ru.kmorozov.library.data.server.storage.ServerStorage;
import ru.kmorozov.onedrive.client.OneDriveItem;
import ru.kmorozov.onedrive.client.OneDriveProvider;

@Component
@Qualifier("remote")
public class ServerGBDOptions extends AbstractServerGBDOptions {

    @Autowired
    @Lazy
    private OneDriveProvider api;

    @Autowired
    @Lazy
    private OneDriveItem root;

    @Override
    public IStorage getStorage() {
        return new ServerStorage(this.api, this.root);
    }

    @Override
    public CtxOptions ctxOptions() {
        return null;
    }

}

File: ru.kmorozov.library.data.server/src/main/java/ru/kmorozov/library/data/server/options/ServerGBDOptions.kt
package ru.kmorozov.library.data.server.options

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.context.annotation.Lazy
import org.springframework.stereotype.Component
import ru.kmorozov.gbd.core.config.IStorage
import ru.kmorozov.library.data.server.storage.ServerStorage
import ru.kmorozov.onedrive.client.OneDriveItem
import ru.kmorozov.onedrive.client.OneDriveProvider

@Component
@Qualifier("remote")
class ServerGBDOptions : AbstractServerGBDOptions() {

    @Autowired
    @Lazy
    private lateinit var api: OneDriveProvider

    @Autowired
    @Lazy
    private val root: OneDriveItem? = null

    override val storage: IStorage
        get() = ServerStorage(api, root)

    override fun serverMode(): Boolean {
        return true
    }

    override val debugEnabled: Boolean
        get() = false
}
